frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/calibration2DHandler/calibration2DHandler.component.tsx

```diff
@@ -15,13 +15,28 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { useEffect } from 'react';
+import { useContext, useEffect } from 'react';
 import { UnityUtil } from '@/globals/unity-util';
+import { CalibrationContext } from '../../calibrationContext';
+import { EMPTY_VECTOR } from '../../calibration.constants';
 
-export const Calibration2DStep = () => {
+export const Calibration2DHandler = () => {
+	const { isCalibrating, step, setIsCalibrating2D, setVector2D } = useContext(CalibrationContext);
+
+	const canCalibrate2D = isCalibrating && step === 1;
 
 	useEffect(() => {
 		UnityUtil.setCalibrationToolMode('Vector');
+
+		return () => {
```

Comment:
(This bug probably comes from a merge) When you reach step 3 and then go back to step 2 or when you recalibrate a calibrated drawing, the 2d vector is gone